poc/four/edfi_client.py

```diff
@@ -0,0 +1,233 @@
+"""
+Ed-Fi API client for CRUD operations on student and enrollment data.
+"""
+import requests
+import logging
+from typing import Dict, List, Optional
+from urllib.parse import urljoin
+from oauth2_client import OAuth2Client
+
+
+class EdFiApiClient:
+    """Client for interacting with Ed-Fi Data Management API."""
+    
+    def __init__(self, base_url: str, oauth_client: OAuth2Client):
+        """
+        Initialize the Ed-Fi API client.
+        
+        Args:
+            base_url: Base URL for Ed-Fi Data Management API
+            oauth_client: OAuth2 client for authentication
+        """
+        self.base_url = base_url.rstrip('/')
```

Comment:
Because the name matches "base URL" in the `.env` - which would be the wrong value - I was worried that this would be incorrect. But when you find the `EdFIApiClient(...)` call, it turns out that it is properly passing the `dataManagementApi` URL, so all is well.